### PR TITLE
Unify test file constants and fixtures

### PR DESCRIPTION
**Pull request by joejuzl**
**Proposed changes**:
- Use fixtures everywhere for test file paths. 
- Combine duplicate nlu and core path fixtures.
- rename test file path fixtures so they are not the same as default file location constant names.

### CODE REVIEW

File: tests/cli/test_rasa_interactive.py [hunk 1]
```diff
@@ -52,15 +51,15 @@ def test_interactive_core_help(run: Callable[..., RunResult]):
 
 
 def test_pass_arguments_to_rasa_train(
-    default_stack_config: Text, monkeypatch: MonkeyPatch
+    stack_config_path, monkeypatch: MonkeyPatch
```
Comment (wochinge) [inline tests/cli/test_rasa_interactive.py:54]: can we please add type annotations for these?
```suggestion
    stack_config_path: Text, monkeypatch: MonkeyPatch
```

File: tests/conftest.py [hunk 2]
```diff
@@ -87,6 +72,105 @@
...
+EXAMPLE_DOMAINS = [
+    "data/test_domains/default_with_mapping.yml",
+    "data/test_domains/default_with_slots.yml",
+    "examples/formbot/domain.yml",
+    "examples/moodbot/domain.yml",
+]
+
+
+@pytest.fixture(scope="session")
+def data_path() -> Text:
```
Comment (wochinge) [inline tests/conftest.py:90]: What about renaming it? `data_path` seems a bit ambiguous to me.
```suggestion
def nlu_as_json_path() -> Text:
```

**Review by wochinge — COMMENTED** [review]
Can we please add the missing type annotations? I guess pycharm omitted them upon rename 😬

**Comment by joejuzl** [comment]
> Can we please add the missing type annotations? I guess pycharm omitted them upon rename 😬
Doing it right now! (and yes it was pycharm...)

File: tests/cli/test_rasa_interactive.py [hunk 3]
```diff
@@ -136,14 +135,14 @@ def test_train_core_called_when_no_model_passed_and_core(
 
 
 def test_no_interactive_without_core_data(
-    default_stack_config: Text, monkeypatch: MonkeyPatch
+    stack_config_path: Text, monkeypatch: MonkeyPatch, nlu_data_path
```
Comment (wochinge) [inline tests/cli/test_rasa_interactive.py:138]:
```suggestion
    stack_config_path: Text, monkeypatch: MonkeyPatch, nlu_data_path: Text
```

File: tests/core/featurizers/test_tracker_featurizer.py [hunk 4]
```diff
@@ -51,11 +49,11 @@ def test_convert_labels_to_ids():
         assert np.all(expected_array == actual_array)
 
 
-def test_featurize_trackers_raises_on_missing_state_featurizer(default_domain: Domain):
+def test_featurize_trackers_raises_on_missing_state_featurizer(domain):
```
Comment (wochinge) [inline tests/core/featurizers/test_tracker_featurizer.py:52]:
```suggestion
def test_featurize_trackers_raises_on_missing_state_featurizer(domain: Domain):
```

File: tests/core/test_agent.py [hunk 5]
```diff
@@ -136,7 +135,7 @@ def test_agent_wrong_use_of_load():
 
 
 async def test_agent_with_model_server_in_thread(
-    model_server: TestClient, default_domain: Domain, unpacked_trained_rasa_model: Text
+    model_server: TestClient, domain, unpacked_trained_rasa_model: Text
```
Comment (wochinge) [inline tests/core/test_agent.py:138]:
```suggestion
    model_server: TestClient, domain: Domain, unpacked_trained_rasa_model: Text
```

File: tests/core/test_nlg.py [hunk 6]
```diff
@@ -74,8 +73,8 @@ async def test_nlg(http_nlg, trained_rasa_model):
     assert response[0] == {"text": "Hey there!", "recipient_id": sender}
 
 
-def test_nlg_endpoint_config_loading():
-    cfg = read_endpoint_config(DEFAULT_ENDPOINTS_FILE, "nlg")
+def test_nlg_endpoint_config_loading(endpoints_path):
```
Comment (wochinge) [inline tests/core/test_nlg.py:76]:
```suggestion
def test_nlg_endpoint_config_loading(endpoints_path: Text):
```

File: tests/core/test_policies.py [hunk 7]
```diff
@@ -101,20 +96,25 @@ def priority(self) -> int:
         return 1
 
     @pytest.fixture(scope="class")
-    def default_domain(self) -> Domain:
-        return Domain.load(DEFAULT_DOMAIN_PATH_WITH_SLOTS)
+    def default_domain(self, domain_path) -> Domain:
```
Comment (wochinge) [inline tests/core/test_policies.py:99]:
```suggestion
    def default_domain(self, domain_path: Text) -> Domain:
```

**Review by wochinge — APPROVED** [review]
Thanks for cleaning this up!

**Pull request merged**